Remove native_functions.yaml dependency from Sort.cu

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #67721
* #66980
* #66979
* #67216
* #66914
* #66913
* #66794
* __->__ #66793

Differential Revision: [D31856100](https://our.internmc.facebook.com/intern/diff/D31856100)

peterbell10 committed:
Remove native_functions.yaml dependency from Sort.cu

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sort.cu"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sort.cu"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sort.cu"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sort.cu"


Differential Revision: [D31856100](https://our.internmc.facebook.com/intern/diff/D31856100)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sort.cu"


Differential Revision: [D31856100](https://our.internmc.facebook.com/intern/diff/D31856100)

[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from Sort.cu"


Differential Revision: [D31856100](https://our.internmc.facebook.com/intern/diff/D31856100)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856100).

dagitses commented on aten/src/ATen/native/cuda/Sort.cpp:
```diff
@@ -0,0 +1,182 @@
+  checkAllSameGPU(__func__, {self_arg, values_arg, indices_arg});
+
+  bool is_non_overlapping_and_dense = self.is_non_overlapping_and_dense();
+  int64_t numel = self.numel();
```
this is unused

dagitses commented on aten/src/ATen/native/cuda/Sort.cpp:
```diff
@@ -0,0 +1,182 @@
+// more than 1024/2048 elements along the selected dimension.
+// Otherwise, we do an inplace bitonic sort (see sortKeyValueInplace).
+bool should_use_small_sort(const TensorBase &self, int64_t dim) {
+  int64_t ndim = self.dim();
```
this is unused

dagitses commented on aten/src/ATen/native/cuda/Sort.h:
```diff
@@ -0,0 +1,19 @@
+namespace at {
+namespace native {
+
+void launch_stable_sort_kernel(const TensorBase &self, int64_t dim, bool descending,
```
can you document this function please since it is newly added.

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sort.cu"


Differential Revision: [D31856100](https://our.internmc.facebook.com/intern/diff/D31856100)

[ghstack-poisoned]

peterbell10 commented on aten/src/ATen/native/cuda/Sort.cpp:
```diff
@@ -0,0 +1,182 @@
+  return nsort <= threshold;
+}
+
+std::vector<int64_t> infer_dense_strides_dim_last(const Tensor & self, int64_t dim);
```
I think it might be related to this comment:
https://github.com/pytorch/pytorch/blob/b00206d4737d1f1e7a442c9f8a1cadccd272a386/aten/src/ATen/native/cuda/UniqueCub.cu#L119-L121

`SortImpl.cu` includes thrust and `Sort.cu` includes cub. Although since this is a host-only function, it would be better to not use `thrust` at all. I'll leave that for a separate PR though.

dagitses replied:
thanks!

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sort.cu"


Differential Revision: [D31856100](https://our.internmc.facebook.com/intern/diff/D31856100)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856100).

Commit:
Update on "Remove native_functions.yaml dependency from Sort.cu"


Differential Revision: [D31856100](https://our.internmc.facebook.com/intern/diff/D31856100)

[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from Sort.cu"


Differential Revision: [D31856100](https://our.internmc.facebook.com/intern/diff/D31856100)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856100).
